cmd/describe/ingress/cmd.go

```diff
@@ -0,0 +1,87 @@
+package ingress
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/ingress"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+const (
+	use     = "ingress"
+	short   = "Show details of the specified ingress within cluster"
+	example = `rosa describe ingress <ingress_id> -c mycluster`
+)
+
+// Regular expression to used to make sure that the identifier given by the
+// user is safe and that it there is no risk of SQL injection:
+var ingressKeyRE = regexp.MustCompile(`^[a-z0-9]{3,5}$`)
+
+func NewDescribeIngressCommand() *cobra.Command {
+	options := NewDescribeIngressUserOptions()
+	cmd := &cobra.Command{
+		Use:     use,
+		Short:   short,
+		Example: example,
+		Run:     rosa.DefaultRunner(rosa.RuntimeWithOCM(), DescribeIngressRunner(options)),
+		Args: func(_ *cobra.Command, argv []string) error {
+			if len(argv) != 1 {
+				return fmt.Errorf(
+					"Expected exactly one command line parameter containing the id of the ingress",
+				)
+			}
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(
+		&options.ingress,
+		"ingress",
+		"",
+		"Ingress of the cluster to target",
+	)
+
+	ocm.AddClusterFlag(cmd)
+	output.AddFlag(cmd)
+	return cmd
+}
+
+func DescribeIngressRunner(userOptions DescribeIngressUserOptions) rosa.CommandRunner {
+	return func(_ context.Context, runtime *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+		options := NewDescribeIngressOptions()
+		if len(argv) == 1 && !cmd.Flag("ingress").Changed {
+			userOptions.ingress = argv[0]
+		} else {
+			err := cmd.ParseFlags(argv)
+			if err != nil {
+				return fmt.Errorf("unable to parse flags: %v", err)
+			}
+		}
+		err := options.Bind(userOptions)
+		if err != nil {
+			return err
+		}
+		ingressKey := options.args.ingress
+		if !ingressKeyRE.MatchString(ingressKey) {
+			return fmt.Errorf(
+				"ingress identifier %q isn't valid: it must contain only letters or digits",
```

Comment:
I think we can move this validation to the bind itself, to make this a little cleaner

Reply:
Oh true, will do